### PR TITLE
[Docs] add missing pipelines from the overview pages and minor fixes

### PR DESCRIPTION
**Pull request by sayakpaul**
This PR adds the missing pipelines from the overview pages that we have under the docs for our pipelines.

### CODE REVIEW

File: docs/source/en/api/pipelines/overview.mdx [hunk 1]
```diff
@@ -70,13 +73,6 @@ available a colab notebook to directly try them out.
 | [stable_diffusion_pix2pix_zero](./stable_diffusion/pix2pix_zero) | [**Zero-shot Image-to-Image Translation**](https://arxiv.org/abs/2302.03027) | Text-Based Image Editing |
 | [stable_diffusion_attend_and_excite](./stable_diffusion/attend_and_excite) | [**Attend and Excite: Attention-Based Semantic Guidance for Text-to-Image Diffusion Models**](https://arxiv.org/abs/2301.13826) | Text-to-Image Generation |
 | [stable_diffusion_self_attention_guidance](./stable_diffusion/self_attention_guidance) | [**Self-Attention Guidance**](https://arxiv.org/abs/2210.00939) | Text-to-Image Generation |
-| [stable_diffusion_image_variation](./stable_diffusion/image_variation) | [**Stable Diffusion Image Variations**](https://github.com/LambdaLabsML/lambda-diffusers#stable-diffusion-image-variations) | Image-to-Image Generation |
```
Comment (patrickvonplaten) [inline docs/source/en/api/pipelines/overview.mdx:73]: I'd leave image variation no?

File: docs/source/en/api/pipelines/overview.mdx [hunk 2]
```diff
@@ -70,13 +73,6 @@ available a colab notebook to directly try them out.
...
-| [stable_diffusion_latent_upscale](./stable_diffusion/latent_upscale) | [**Stable Diffusion Latent Upscaler**](https://twitter.com/StabilityAI/status/1590531958815064065) | Text-Guided Super Resolution Image-to-Image |
```
Comment (patrickvonplaten) [inline docs/source/en/api/pipelines/overview.mdx:74]: would also leave latent upscale

File: docs/source/en/api/pipelines/overview.mdx [hunk 3]
```diff
@@ -70,13 +73,6 @@ available a colab notebook to directly try them out.
...
-| [stable_diffusion_2](./stable_diffusion_2/) | [**Stable Diffusion 2**](https://stability.ai/blog/stable-diffusion-v2-release) | Text-to-Image Generation | 
-| [stable_diffusion_2](./stable_diffusion_2) | [**Stable Diffusion 2**](https://stability.ai/blog/stable-diffusion-v2-release) | Text-Guided Image Inpainting | 
-| [stable_diffusion_2](./stable_diffusion_2) | [**Stable Diffusion 2**](https://stability.ai/blog/stable-diffusion-v2-release) | Depth-to-Image Text-Guided Generation |
-| [stable_diffusion_2](./stable_diffusion_2) | [**Stable Diffusion 2**](https://stability.ai/blog/stable-diffusion-v2-release) | Text-Guided Super Resolution Image-to-Image |
-| [stable_diffusion_safe](./stable_diffusion_safe) | [**Safe Stable Diffusion**](https://arxiv.org/abs/2211.05105) | Text-Guided Generation | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ml-research/safe-latent-diffusion/blob/main/examples/Safe%20Latent%20Diffusion.ipynb)
```
Comment (patrickvonplaten) [inline docs/source/en/api/pipelines/overview.mdx:79]: would also leave this

File: docs/source/en/api/pipelines/overview.mdx (hunk 1 above)
Comment (sayakpaul) [reply docs/source/en/api/pipelines/overview.mdx:73]: Why though? It's already present in the overview section of the dedicated page we have for Stable Diffusion. For better visibility and discoverability?

File: docs/source/en/api/pipelines/overview.mdx (hunk 2 above)
Comment (sayakpaul) [reply docs/source/en/api/pipelines/overview.mdx:74]: Same as https://github.com/huggingface/diffusers/pull/3795#discussion_r1231045991

File: docs/source/en/api/pipelines/overview.mdx (hunk 3 above)
Comment (sayakpaul) [reply docs/source/en/api/pipelines/overview.mdx:79]: Same as https://github.com/huggingface/diffusers/pull/3795#discussion_r1231045991.

**Comment by sayakpaul** [comment]
@patrickvonplaten @yiyixuxu I addressed your concerns. PTAL now.

**Review by patrickvonplaten — APPROVED** [review]
Cool!

**Pull request merged**